File: Esports-Application/src/Entity/Post.php
<?php

namespace App\Entity;

use App\Repository\PostRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\HttpFoundation\File\File;
use Symfony\Component\HttpFoundation\File\UploadedFile;
use Vich\UploaderBundle\Mapping\Annotation as Vich;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups;
use Vangrg\ProfanityBundle\Entity\Profanity;
use Zend\Json\Expr;
use ProfanityAssert\ProfanityCheck;

/**
 * @ORM\Entity(repositoryClass=PostRepository::class)
 * @Vich\Uploadable
 */
class Post
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     * @Groups("post:read")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     @Assert\Regex(
     *     pattern="/\d/",
     *     match=false,
     *     message="le sujet doit comporter que des lettres"
     * )
     * @Groups("post:read")
     */
    private $sujet;

    /**
     * @ORM\Column(type="string", length=255)
    @Assert\Length(
     *     min = 10,
     *     max = 255,
     *     minMessage= "la description  est trop court",
     *     maxMessage="la description est trop long"
     * )
     * @Groups("post:read")

     */
    private $description;

    /**
     * @ORM\Column(type="integer")

     * @Groups("post:read")
     */
    private $nbr_jaime;



    /**
     * @ORM\Column(type="string", length=255, nullable=true)
     * @Groups("post:read")
     */
    private $image;

    /**
     * @ORM\Column(type="string", length=255)
     * @Groups("post:read")
     */
    private $nom_user;



    /**
     * @ORM\Column(type="datetime")
     * @Groups("post:read")
     */
    private $date_p;
    /**

     * @Vich\UploadableField(mapping="post_image", fileNameProperty="image")
     * @Groups("post:read")
     *
     * @var File|null
     */
    private $imageFile;

    /**
     * @ORM\Column(type="string", length=255)
    @Assert\Regex(
     *     pattern="/\d/",
     *     match=false,
     *     message="la communauté doit comporter que des lettres"
     * )
     * @Groups("post:read")
     */
    private $communaute;

    /**
     * @ORM\OneToMany(targetEntity=Commentaire::class, mappedBy="post", cascade={"remove"})
     * @ORM\JoinColumn(onDelete="CASCADE")

     */
    private $commentaires;

    /**
     * @ORM\Column(type="string", length=255, nullable=true)
     */
    private $analysePo;

    public function __construct()
    {
        $this->commentaires = new ArrayCollection();
    }













    public function getId(): ?int
    {
        return $this->id;
    }

    public function getSujet(): ?string
    {
        return $this->sujet;
    }

    public function setSujet(string $sujet): self
    {
        $this->sujet = $sujet;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getNbrJaime(): ?int
    {
        return $this->nbr_jaime;
    }

    public function setNbrJaime(int $nbr_jaime): void
    {
        $this->nbr_jaime = $nbr_jaime;


    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): self
    {
        $this->image = $image;

        return $this;
    }

    public function getNomUser(): ?string
    {
        return $this->nom_user;
    }

    public function setNomUser(string $nom_user): self
    {
        $this->nom_user = $nom_user;

        return $this;
    }

    public function getCommunaute(): ?string
    {
        return $this->communaute;
    }

    public function setCommunaute(string $communaute): self
    {
        $this->communaute = $communaute;

        return $this;
    }

    public function getDateP(): ?\DateTimeInterface
    {
        return $this->date_p;
    }

    public function setDateP(\DateTimeInterface $date_p): self
    {
        $this->date_p = $date_p;

        return $this;
    }
    public function setImageFile(?File $imageFile = null): self
    {
        $this->imageFile = $imageFile;

       return $this;
    }

    public function getImageFile(): ?File
    {
        return $this->imageFile;
    }

    /**
     * @return Collection<int, Commentaire>
     */
    public function getCommentaires(): Collection
    {
        return $this->commentaires;
    }

    public function addCommentaire(Commentaire $commentaire): self
    {
        if (!$this->commentaires->contains($commentaire)) {
            $this->commentaires[] = $commentaire;
            $commentaire->setPost($this);
        }

        return $this;
    }

    public function removeCommentaire(Commentaire $commentaire): self
    {
        if ($this->commentaires->removeElement($commentaire)) {
            // set the owning side to null (unless already changed)
            if ($commentaire->getPost() === $this) {
                $commentaire->setPost(null);
            }
        }

        return $this;
    }

    public function getAnalysePo(): ?string
    {
        return $this->analysePo;
    }

    public function setAnalysePo(?string $analysePo): self
    {
        $this->analysePo = $analysePo;

        return $this;
    }






}

File: PiJava/src/edu/esprit/entities/Post.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package edu.esprit.entities;

import java.sql.Date;

/**
 *
 * @author USER
 */
public class Post {
    private int id;
    private String sujet;
    private String description;
    private String communaute;
    private String nom_user;
    private Date date_p;
    private String image;
     private int nbr_jaime;
          private int liked;
          private int badlevel;

    
static int test;

    public static int getTest() {
        return test;
    }

    public static void setTest(int test) {
        Post.test = test;
    }
    public Post() {
    }

    public Post(String description, String communaute, String nom_user, String image, int badlevel) {
        this.description = description;
        this.communaute = communaute;
        this.nom_user = nom_user;
        this.image = image;
        this.badlevel = badlevel;
    }
    

    public Post(String sujet, String description, String communaute, String nom_user, String image, int badlevel) {
        this.sujet = sujet;
        this.description = description;
        this.communaute = communaute;
        this.nom_user = nom_user;
        this.image = image;
        this.badlevel = badlevel;
    }

    public Post(int idPost, String sujet, String description, String communaute, String nom_user,  String image, int nbr_jaime) {
        this.id = idPost;
        this.sujet = sujet;
        this.description = description;
        this.communaute = communaute;
        this.nom_user = nom_user;
        this.image = image;
        this.nbr_jaime = nbr_jaime;
    }

    public Post(int id, String sujet, String description, String communaute, String nom_user, Date date_p, String image, int nbr_jaime) {
        this.id = id;
        this.sujet = sujet;
        this.description = description;
        this.communaute = communaute;
        this.nom_user = nom_user;
        this.date_p = date_p;
        this.image = image;
        this.nbr_jaime = nbr_jaime;
    }

    public Post(String sujet, String description, String communaute, String nom_user, String image) {
        this.sujet = sujet;
        this.description = description;
        this.communaute = communaute;
        this.nom_user = nom_user;
        this.image = image;
    }

    public Post(int id, String sujet, String description, String communaute, String nom_user, Date date_p, String image, int nbr_jaime, int liked) {
        this.id = id;
        this.sujet = sujet;
        this.description = description;
        this.communaute = communaute;
        this.nom_user = nom_user;
        this.date_p = date_p;
        this.image = image;
        this.nbr_jaime = nbr_jaime;
        this.liked = liked;
    }

    public Post(int id, String sujet, String description, String communaute, String nom_user, Date date_p, String image, int nbr_jaime, int liked, int badlevel) {
        this.id = id;
        this.sujet = sujet;
        this.description = description;
        this.communaute = communaute;
        this.nom_user = nom_user;
        this.date_p = date_p;
        this.image = image;
        this.nbr_jaime = nbr_jaime;
        this.liked = liked;
        this.badlevel = badlevel;
    }

    

   

    

    
    public int getIdPost() {
        return id;
    }

    public void setIdPost(int idPost) {
        this.id = idPost;
    }

    public String getSujet() {
        return sujet;
    }

    public void setSujet(String sujet) {
        this.sujet = sujet;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getCommunaute() {
        return communaute;
    }

    public void setCommunaute(String communaute) {
        this.communaute = communaute;
    }

    public String getNom_user() {
        return nom_user;
    }

    public void setNom_user(String nom_user) {
        this.nom_user = nom_user;
    }

    public Date getDate_p() {
        return date_p;
    }

    public void setDate_p(Date date_p) {
        this.date_p = date_p;
    }

    public int getNbr_jaime() {
        return nbr_jaime;
    }

    public void setNbr_jaime(int nbr_jaime) {
        this.nbr_jaime = nbr_jaime;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

  

    public int getLiked() {
        return liked;
    }

    public void setLiked(int liked) {
        this.liked = liked;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getBadlevel() {
        return badlevel;
    }

    public void setBadlevel(int badlevel) {
        this.badlevel = badlevel;
    }

    @Override
    public String toString() {
        return "Post{" + "id=" + id + ", sujet=" + sujet + ", description=" + description + ", communaute=" + communaute + ", nom_user=" + nom_user + ", date_p=" + date_p + ", image=" + image + ", nbr_jaime=" + nbr_jaime + ", liked=" + liked + ", badlevel=" + badlevel + '}';
    }
    
}
